test/libsolidity/semanticTests/viaYul/memory_struct_allow.sol

```diff
@@ -0,0 +1,21 @@
+contract C {
+    struct S {
+        uint256 a;
+        uint256 b;
+    }
+
+    function f() public pure returns (uint256 a, uint256 b){
+        assembly {
+            // Make free memory dirty to check that the struct allocation cleans it up again.
+            let freeMem := mload(0x40)
+            mstore(freeMem, 42)
+            mstore(add(freeMem, 32), 42)
+        }
+        S memory s;
```

Comment:
Why is the struct allocation expected to clean up previous two mstores (as per the comment above)?

Reply:
Ah I think I answered my question above :-)

tl;dr: Uninitialized allocation should clear previously stored memory values